kubernetes/kubernetes.go

```diff
@@ -234,7 +256,32 @@ func isCustomS3UploadEnvsSet(ka *KubeAgentConfig) bool {
 	}
 	log.Infof("Detected custom S3 bucket location and S3 bucket region. "+
 		"Will upload collected metrics to %s in the aws region %s", ka.CustomS3UploadBucket, ka.CustomS3Region)
-	return true
+	return true, "aws"
+}
+
+// isCustomAzureUploadEnvsSet checks to see if the agent has custom Azure credentials needed for uploads
+// if these vars are found, set the env vars needed for the azure client. If not set, default upload to Apptio S3
+func isCustomAzureUploadEnvsSet(ka *KubeAgentConfig) (bool, string) {
+	if ka.CustomAzureUploadBlobContainerName == "" && ka.CustomAzureBlobURL == "" && ka.CustomAzureClientID == "" &&
+		ka.CustomAzureClientSecret == "" && ka.CustomAzureTenantID == "" {
```

Comment:
Could break out the giant equals blob and set it to a bool rather than duplicating it.

Reply:
can't break them out as the first check is and operators and the second check is or operators